### PR TITLE
Mapping module deployment outputs to github action output

### PR DESCRIPTION
**Pull request by eriqua**
# Change

Mapping the module deployment output to a validation deployment github action output variable.
The key-value pairs are converted to a single json string.
This small change will allow us to leverage module deployment outputs from any workflow using the action e.g. the dependency pipeline or solutions using a pipeline orchestrated approach. 
Can also be leveraged to use deployment outputs as input for subsequent jobs.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [] I'm sure there are no other open Pull Requests for the same update/change
- [] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

**Review by MrMCake — CHANGES_REQUESTED** [review]
Any changes required for the AzureDevOps templates too?

**Comment by eriqua** [comment]
> Any changes required for the AzureDevOps templates too?

Agree, working on it

File: .azuredevops/pipelineTemplates/module.jobs.deploy.yml [hunk 1]
```diff
@@ -231,6 +231,7 @@ jobs:
           #---------------------
           - task: AzurePowerShell@5
             displayName: 'Deploy [${{ deploymentBlock.path }}] via connection [${{ parameters.serviceConnection }}]'
+            name: 'DeployModule'
```
Comment (MrMCake) [inline .azuredevops/pipelineTemplates/module.jobs.deploy.yml:234]: Any reason to add the 'id/name'?

Comment (MrMCake) [reply .azuredevops/pipelineTemplates/module.jobs.deploy.yml:234]: It's not referenced

Comment (eriqua) [reply .azuredevops/pipelineTemplates/module.jobs.deploy.yml:234]: The id is needed to allow accessing the output from another stage. E.g.

```
- stage: stageB
  dependsOn: stageA
  jobs:
  - job: stageBjob
    variables:
        varFromStageA: $[ stageDependencies.stageA.stageAjob.outputs['DeployModule.deploymentOutput'] ]
```

**Review by MrMCake — APPROVED** [review]

**Pull request merged**